fieldpath/set_test.go

```diff
@@ -447,6 +453,71 @@ func TestSetIntersectionDifference(t *testing.T) {
 	})
 }
 
+func TestSetRecursiveDifference(t *testing.T) {
+	table := []struct {
+		name   string
+		a      *Set
+		b      *Set
+		expect *Set
```

Comment:
Can you add in the expectation for the regular difference? I'm actually not following how this is different?

Reply:
I added both differences to the test and added the test i forgot for both.